blivet/osinstall.py

```diff
@@ -1465,176 +1470,22 @@ def _update_custom_storage_ksdata(self):
             parent = getattr(self.ksdata, list_attr)
             parent.dataList().append(data)
 
-        self.fsset = FSSet(self.devicetree)
+        self.config = StorageDiscoveryConfig()
+        self.autopart_type = AUTOPART_TYPE_LVM
 
-    def reset(self, cleanup_only=False):
-        Blivet.reset(self, cleanup_only)
+        self.__luks_devs = {}
 
+        # these will both be empty until our reset method gets called
+        # instantiate our own devicetree here to override the default created
+        # in Blivet so that protected_dev_specs gets handled
+        self.devicetree = DeviceTree(passphrase=self.encryption_passphrase,
+                                     luks_dict=self.__luks_devs,
+                                     ignored_disks=self.ignored_disks,
+                                     exclusive_disks=self.exclusive_disks,
+                                     disk_images=self.disk_images,
+                                     protected_dev_specs=self.config.protected_dev_specs)
```

Comment:
We talked about moving the protected device handling out of `DeviceTree` completely, since all we do there is set a simple attribute (`device.protected`) anyway.

We also talked about refactoring such that `InstallerSorage.reset` include a call to `super().reset(cleanup_only=cleanup_only)` instead of duplicating that code.

As far as `Blivet` having its own configuration class/instance, I do like the idea of reducing clutter somewhat, but I'm not sure it's warranted just for ignored_disks, exclusive_disks, and disk_images.